common/flatpak-dir.c

```diff
@@ -3993,24 +4002,33 @@ flatpak_dir_prepare_resolve_p2p_refs_helper (FlatpakDir             *self,
     {
       FlatpakDirResolve *resolve = resolves->pdata[i];
       const char *latest_rev = NULL;
+      g_autoptr(GVariant) commit_data = NULL;
+      guint64 local_timestamp, remote_timestamp;
 
-      latest_rev = find_latest_p2p_result (results, &resolve->collection_ref);
+      latest_rev = find_latest_p2p_result (results, &resolve->collection_ref, &remote_timestamp);
       resolve->latest_remote_commit = g_strdup (latest_rev);
 
-      if (resolve->local_commit == NULL)
+      if (resolve->local_commit == NULL || latest_rev == NULL)
         continue;
 
-      if (g_strcmp0 (latest_rev, resolve->local_commit) == 0)
-        {
-          g_autoptr(GVariant) commit_data = NULL;
+      if (!ostree_repo_load_commit (state->child_repo, resolve->local_commit, &commit_data, NULL, NULL))
+        return FALSE;
```

Comment:
We're returning FALSE here without setting error

Reply:
Right, but that's the issue fixed by the following commit. The reason it's split across two commits is that it's two commits in the Endless flatpak repo, and these are cherry-picks.

Reply:
And actually this commit isn't introducing this bug, just moving it from a few lines below.